x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/fleet_server_upgrade_modal.tsx

```diff
@@ -37,6 +45,58 @@ export const FleetServerUpgradeModal: React.FunctionComponent<Props> = ({ onClos
   const isCloud = !!cloud?.cloudId;
 
   const [checked, setChecked] = useState(false);
+  const [isAgentsAndPoliciesLoaded, setAgentsAndPoliciesLoaded] = useState(false);
+
+  // Check if an other agent than the fleet server is already enrolled
+  useEffect(() => {
+    async function check() {
+      try {
+        const agentPoliciesAlreadyChecked: { [k: string]: boolean } = {};
+
+        const res = await sendGetAgents({
+          page: 1,
+          perPage: 10,
+          showInactive: false,
+        });
+
+        if (res.error) {
+          throw res.error;
+        }
+
+        for (const agent of res.data?.list ?? []) {
+          if (agent.policy_id) {
```

Comment:
Could this check be inverted as a guard clause and combined with the other `if` statement to reduce nesting for the bulk of this logic?

```ts
if (!agent.policy_id || agentPoliciesAlreadyChecked[agent.policy_id]) { 
  continue
}

agentPoliciesAlreadyChecked[agent.policy_id] = true;
const policyRes = await sendGetOneAgentPolicy(agent.policy_id);
//...
```